hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -129,10 +153,28 @@ public HandleContextImpl(
         this.serviceScopeLookup = requireNonNull(serviceScopeLookup, "serviceScopeLookup must not be null");
         this.blockRecordInfo = requireNonNull(blockRecordInfo, "blockRecordInfo must not be null");
         this.recordCache = requireNonNull(recordCache, "recordCache must not be null");
+        this.feeManager = requireNonNull(feeManager, "feeManager must not be null");
+        this.userTransactionConsensusTime =
+                requireNonNull(userTransactionConsensusTime, "userTransactionConsensusTime must not be null");
+        this.feeCalculatorCreator = (subType) -> feeManager.createFeeCalculator(
+                txInfo, payerKey, verifier.numSignaturesVerified(), userTransactionConsensusTime, subType);
 
         final var serviceScope = serviceScopeLookup.getServiceName(txBody);
         this.writableStoreFactory = new WritableStoreFactory(stack, serviceScope);
         this.serviceApiFactory = new ServiceApiFactory(stack);
+
+        final var tokenApi = this.serviceApiFactory.getApi(TokenServiceApi.class);
+        this.feeAccumulator = new FeeAccumulator() {
```

Comment:
I'd prefer a regular class here.

Reply:
Noted. I think we can clean that up in a future PR.